### DISCUSSION TITLE
npm error

### DISCUSSION THREAD

**Discussion by SoumyadeepChatterjee96** [termux/termux-app#2380] [Q&A]
So I have been trying to install node red on my android device however this error comes up no matter what I try and I am using the fdroid version (if that helps) on my android 11 lineage os and I am a noob right now trying to work things out once at a time so please someone help me understand the error and how to resolve this 
$ npm i -g --unsafe-perm node-red
npm ERR! Exit handler never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/cli/issues>

npm ERR! A complete log of this run can be found in:
npm ERR!     /data/data/com.termux/files/home/.npm/_logs/2021-11-05T09_09_31_968Z-debug.log

**Comment by tomap** [comment]
Did you follow those steps ?
https://nodered.org/docs/getting-started/android

**Reply by ganeshkamal-ar** [reply]
I also have the same error while installing angular js

**Comment by thunder-coding** [comment]
Consider using `yarn` instead of `npm`. `npm` has various issues on Termux. 

Helpful upstream link: https://github.com/npm/statusboard/issues/395

**Reply by ghost** [reply]
```js
~ $ npm install -g npm@8.1.3

changed 14 packages, and audited 219 packages in 9s

10 packages are looking for funding
  run `npm fund` for details

3 moderate severity vulnerabilities

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
```
```js
~ $ npm install discord.js
npm ERR! code ENOTEMPTY
npm ERR! syscall rename
npm ERR! path /data/data/com.termux/files/home/node_modules/agent-base
npm ERR! dest /data/data/com.termux/files/home/node_modules/.agent-base-GPa1owoj
npm ERR! errno -39
npm ERR! ENOTEMPTY: directory not empty, rename '/data/data/com.termux/files/home/node_modules/agent-base' -> '/data/data/com.termux/files/home/node_modules/.agent-base-GPa1owoj'

npm ERR! A complete log of this run can be found in:
npm ERR!     /data/data/com.termux/files/home/.npm/_logs/2021-11-16T06_52_35_189Z-debug.log
```
